wetube-api/src/main/kotlin/com/leeparkim/wetube/presentation/auth/SocialOauthController.kt

```diff
@@ -0,0 +1,28 @@
+package com.leeparkim.wetube.presentation.auth
+
+import com.leeparkim.wetube.application.auth.OAuthApplicationService
+import com.leeparkim.wetube.domain.user.SocialType
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.ResponseBody
+
+
+@Controller
+class SocialOauthController(private val oAuthService: OAuthApplicationService) {
+
+    @GetMapping("/signin/social/{socialType}/form")
+    fun socialLoginRedirect(@PathVariable(name="socialType") socialTypeString: String): String {
+        val redirectUrl = oAuthService.getRedirectUrl(socialTypeString);
+
+        return "redirect:$redirectUrl"
+    }
+
+    @ResponseBody
+    @GetMapping("/signin/social/{socialType}/form/callback")
+    fun socialCallback(@PathVariable(name="socialType") socialTypeString: String, code: String): String? {
+        val userId = oAuthService.findSocialUserId(socialTypeString, code)?: return null
+        return userId
```

Comment:
controller 단에서 null을 내보는것보단 Error를 뱉는게 나아보여요. 예를들어
```suggestion
        return oAuthService.findSocialUserId(socialTypeString, code) ?: throw IllegalStateException("~~")   
```
식으로 변경하고 IllgalStateException에 대한 예외를 공통적으로 처리하는 로직을 둬서 RestApi에 맞는 에러를 내뱉는 식으로여